include/quicr/message_buffer.h

```diff
@@ -178,19 +178,18 @@ class MessageBuffer
 
   /**
    * @brief Writes QUICR integral types to the buffer in NBO.
-   * @tparam T A type satisfying quicr::is_integral.
+   * @tparam T An unsigned integral or a quicr::Name.
    * @param value The message to be written.
    * @returns The MessageBuffer that was written to.
    */
-  template<typename T, typename = std::enable_if_t<quicr::is_integral_v<T>, T>>
+  template<UnsignedOrName T>
   inline MessageBuffer& operator<<(T value)
   {
     value = swap_bytes(value);
-    value_type* val_ptr = reinterpret_cast<value_type*>(&value);
 
     const auto length = _buffer.size();
     _buffer.resize(length + sizeof(T));
-    std::memcpy(_buffer.data() + length, val_ptr, sizeof(T));
+    std::memcpy(_buffer.data() + length, &value, sizeof(T));
```

Comment:
Rather than bothering with endianness, you should just deal with the integral type semantically, and do the big-endian encoding yourself.

```c++
  auto tail = span_type(buffer).subspan(length);
  const auto int_size = tail.size();
  for (int i = 0; i < int_size; i++) {
    tail[i] = static_cast<uint8_t>(value >> unsigned(8 * (int_size - i - 1)));
  }
```

Likewise in `operator>>`

Reply:
We are doing it ourselves in swap_bytes. And it's compile time evaluated, so it doesn't cost anything to check endianness.